src/Microsoft.DotNet.XHarness.InstrumentationBase.Xunit/DefaultAndroidEntryPoint.cs

```diff
@@ -0,0 +1,98 @@
+        private const string ExcludeCategoriesFileArgumentName = "exclude-categories-file";
+        private const string ExcludeMethodArgumentName = "exclude-method";
+        private const string IncludeMethodArgumentName = "include-method";
+        private const string ExcludeClassArgumentName = "exclude-class";
+        private const string IncludeClassArgumentName = "include-class";
+        private readonly string _resultsPath;
+        private readonly string _excludeCategoriesDir;
+        private readonly string _excludeCategoriesFile;
+        private readonly Dictionary<string, string> _parsedArguments;
+
+        protected override string IgnoreFilesDirectory => _excludeCategoriesDir;
+        protected override string IgnoredTraitsFilePath => _excludeCategoriesFile;
+
+        public DefaultAndroidEntryPoint(Dictionary<string, string> bundle)
+        {
+            _parsedArguments = bundle;
+
+            // use default name for test results file
+            _parsedArguments.TryAdd(ResultsFileArgumentName, "TestResults.xml");
+
+            if (!Directory.Exists(_parsedArguments[ResultsFileArgumentPath]))
+            {
+                Directory.CreateDirectory(_parsedArguments[ResultsFileArgumentPath]);
+            }
+
+            _resultsPath = Path.Combine(_parsedArguments[ResultsFileArgumentPath], _parsedArguments[ResultsFileArgumentName]);
+            _excludeCategoriesDir = _parsedArguments.GetValueOrDefault(ExcludeCategoriesDirArgumentName);
+            _excludeCategoriesFile = _parsedArguments.GetValueOrDefault(ExcludeCategoriesFileArgumentName);
+        }
+
+        protected override bool LogExcludedTests => true;
+
+        public override TextWriter? Logger => null;
+
+        public override string TestsResultsFinalPath => _resultsPath;
+
+        protected override int? MaxParallelThreads => System.Environment.ProcessorCount/2;
+
+        protected override IDevice? Device => null;
+
+        public IEnumerable<Assembly> Tests { get; set; } = new List<Assembly>();
+        protected override IEnumerable<TestAssemblyInfo> GetTestAssemblies()
+        {
+            foreach (var assembly in Tests)
+            {
+                yield return new TestAssemblyInfo(assembly, assembly.Location);
+            }
+        }
+        protected override void TerminateWithSuccess()
+        {
+        }
+
+        private void ConfigureFilters(string? filter, Action<string, bool> filterMethod, bool isExcluded)
+        {
+            if (filter != null)
+            {
+                foreach (var f in filter.Split(' '))
+                {
+                    filterMethod(f, isExcluded);
+                }
+            }
+        }
+
+        protected override TestRunner GetTestRunner(LogWriter logWriter)
+        {
+            var testRunner = base.GetTestRunner(logWriter);
+
+            Tuple<string, Action<string, bool>, bool>[] filters =
```

Comment:
FYI, in C# 8.0+, you can use tuples without the Tuple class, you can even name the items.

The initialization would be:
```csharp
(string, Action<string, bool>, bool)[] filters
```
or with named version:
```csharp
(string Filter, Action<string, bool> FilterMethod, bool IsExcluded)[] filters
```
And the initialization would then be also shorter:
```csharp
(_parsedArguments.GetValueOrDefault(ExcludeMethodArgumentName), testRunner.SkipMethod, true),
```
Below, you would then do:
```csharp
ConfigureFilters(t.Filter, t.FilterMethod, t.IsExcluded);
```

Reply:
Or, please also consider creating a private class/record inside of this class:
```csharp
private class FilterDefinition { ...
```

And then use it here and in the ConfigureFilters method.

Reply:
Although, this ConfigureFilters is so short it could be inlined as well without problems too.

Reply:
Cool, I was looking for something like this initialization, thanks!

Reply:
done